pakyow-core/spec/unit/middleware/req_path_normalizer_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+require 'core/middleware/req_path_normalizer'
+
+describe Pakyow::Middleware::ReqPathNormalizer do
+  let :app do
+    double
+  end
+
+  let :instance do
+    Pakyow::Middleware::ReqPathNormalizer.new(app)
+  end
+
+  let :env do
+    {
+      'PATH_INFO' => path,
+      'REQUEST_METHOD' => method
+    }
+  end
+
+  let :path do
+    '//test_page'
+  end
+
+  let :normalized_path do
+    instance.normalize_path(env['PATH_INFO'])
+  end
+
+  let :method do
+    'GET'
+  end
+
+  before do
+    allow(Pakyow).to receive(:app).and_return(app)
+    allow(app).to receive(:call)
+  end
+
+  describe '#call' do
+    it 'exists' do
+      expect(instance).to respond_to(:call)
+    end
+
+    it 'accepts one arg' do
+      expect(instance.method(:call).arity).to eq(1)
+    end
+
+    context 'path contains //' do
+      it 'issues a 301 redirect with the normalized path' do
+        expect_any_instance_of(Pakyow::CallContext).to receive(:redirect).with(normalized_path, 301)
```

Comment:
:+1: I'll do that and let you know how it goes :)
One more thing, I'm not sure what I should be expecting in this spec, as you can see line 49 has `instance.call(env)` which I don't will end up verifying what I need, any thoughts on that?